executive_logic/executive_logic.py

```diff
@@ -1,36 +1,204 @@
+        self.__query_server(MessageType.END_GAME, [winner_id])  # Tell server who won, and to end game
+        self.is_game_running = False
+
+    def __next_player(self, curr_player_id):
+        """
+        Increments ID of current player (self.curr_player_id). In a 3-player game, cycles between 0, 1, and 2.
+        :return: void
+        """
+
+        if curr_player_id is None:
+            return 0
+        else:
+            return (curr_player_id + 1) % len(self.game_server.players)
+
+    def __spin_wheel(self):
+        """
+        Called whenever the wheel must be spun. Increments spin counter and returns spin result.
+        :return: Result of wheel spin
+        """
         self.num_spins += 1
-        return self.num_spins
+        return self.wheel.get_spin_result()
 
-    def getRound(self):
-        return self.round
+    def __query_server(self, command: MessageType, args: list):
+        """
+        Called whenever information must be obtained from the server (e.g. asking the user for something).
+        Stalls until a response is received. Will only accept a response of the type that was requested.
+        :param command: (MessageType) The command to be executed, like "JEOPARDY_CATEGORY" or "PLAYERS_CHOICE"
+        :param args: (list) Any additional arguments that must be sent to the server
+        :return: void
+        """
+        self.query_response = Message(MessageType.EMPTY, [])  # Clear server's previous response
+        self.query_status = QueryStatus.SERVER_TO_CLIENT  # Tell server we want to query client
+        self.server_message = Message(command, args)  # Store the query in self.server_message for server to read
 
-    def whoseTurn(self):
-        return "Wil"
+        while self.query_response.code is not command:  # Wait for a response of the right type from the server
+            pass
 
-    def updateTurn(self):
-        # return the next player
-        return "Luis"
+    def store_query(self, command: MessageType, args: list):
+        """
+        Called by the server to return the result of a query from __query_server.
+        :param command: (MessageType) The command that was executed, like "JEOPARDY_CATEGORY" or "PLAYERS_CHOICE"
+        :param args: (list) Any information that must be returned to ExecLogic
+        :return: void
+        """
+        self.query_response = Message(command, args)  # Store server response in self.query_response
 
-    def selectRandOpponent(self):
-        return "Opponent"
+    def __update_scores_tokens_spins(self):
+        """
+        Called after a game value is updated that should be reflected in the UI display.
+        Sends server the current player scores, player tokens, and number of spins.
+        :return: void
+        """
+        self.__query_server(MessageType.UPDATE_SCORES,
+                            [self.score_keeper.get_scores(), self.score_keeper.get_tokens(), self.num_spins])
```

Comment:
I think we should send the number of spins remaining, that way the Client/UI doesn't also have to keep up with MAX_SPINS.